matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/state/DefaultStateService.kt

```diff
@@ -155,12 +158,14 @@ internal class DefaultStateService @AssistedInject constructor(@Assisted private
     }
 
     override suspend fun updateAvatar(avatarUri: Uri, fileName: String) {
-        val response = fileUploader.uploadFromUri(avatarUri, fileName, MimeTypes.Jpeg)
-        sendStateEvent(
-                eventType = EventType.STATE_ROOM_AVATAR,
-                body = mapOf("url" to response.contentUri),
-                stateKey = ""
-        )
+        withContext(coroutineDispatchers.io) {
+            val response = fileUploader.uploadFromUri(avatarUri, fileName, MimeTypes.Jpeg)
+            sendStateEvent(
```

Comment:
I think this could stay outside of the `withContext(coroutineDispatchers.io)` block.

Reply:
This method is used when we update the picture in the room settings. 
It's not working for me in release (i have a popup : "Sorry, an error occurred")but with this fix it's working.

Reply:
Ok, so please do it, it will be cleaner. Thanks!